wvd-templates/DSC/Script-TestRegisterSessionHost.ps1

```diff
@@ -0,0 +1,97 @@
+﻿<#
+
+.SYNOPSIS
+Test if Session Host was registered to the Host Pool.
+
+.DESCRIPTION
+The supported Operating Systems Windows Server 2016.
+
+.ROLE
+Readers
+
+#>
+param(
+    [Parameter(mandatory = $true)]
+    [string]$RDBrokerURL,
+
+    [Parameter(mandatory = $true)]
+    [string]$definedTenantGroupName,
+
+    [Parameter(mandatory = $true)]
+    [string]$TenantName,
+
+    [Parameter(mandatory = $true)]
+    [string]$HostPoolName,
+
+    [Parameter(mandatory = $true)]
+    [PSCredential]$TenantAdminCredentials,
+
+    [Parameter(mandatory = $false)]
+    [string]$isServicePrincipal = "False",
+
+    [Parameter(mandatory = $false)]
+    [AllowEmptyString()]
+    [string]$AadTenantId = "",
+
+    [Parameter(mandatory = $false)]
+    [string]$RDPSModSource = 'attached'
+)
+
+$ScriptPath = [System.IO.Path]::GetDirectoryName($PSCommandPath)
+
+# Dot sourcing Functions.ps1 file
+. (Join-Path $ScriptPath "Functions.ps1")
+
+# Setting ErrorActionPreference to stop script execution when error occurs
+$ErrorActionPreference = "Stop"
+
+write-log -message 'Script being executed: Test if Session Host is registered to the Host Pool'
+
+Write-Log -Message "Check if RD Infra registry exists"
+$RDInfraReg = Get-ItemProperty -Path "Registry::HKEY_LOCAL_MACHINE\SOFTWARE\Microsoft\RDInfraAgent" -ErrorAction SilentlyContinue
+if (!$RDInfraReg) {
+    Write-Log -Message "Session Host is not registered to the Host Pool (RD Infra registry missing)"
+    return $false
+}
+Write-Log -Message "RD Infra registry exists"
+
+Write-Log -Message "Check RD Infra registry to see if RD Agent is registered"
+if ($RDInfraReg.RegistrationToken -ne '') {
+    Write-Log -Message "Session Host is not registered to the Host Pool (RegistrationToken in RD Infra registry is not empty: '$($RDInfraReg.RegistrationToken)')"
+    return $false
+}
+if ($RDInfraReg.IsRegistered -ne 1) {
+    Write-Log -Message "Session Host is not registered to the Host Pool (Value of 'IsRegistered' in RD Infra registry is not 1: $($RDInfraReg.IsRegistered))"
+    return $false
+}
+
+Write-Log -Message "Accoring to RD Infra registry, RD Agent appears to be registered"
+
+# Testing if it is a ServicePrincipal and validade that AadTenant ID in this case is not null or empty
+ValidateServicePrincipal -IsServicePrincipal $isServicePrincipal -AADTenantId $AadTenantId
+
+ImportRDPSMod -Source $RDPSModSource -ArtifactsPath $ScriptPath
+
+# Authenticating to Windows Virtual Desktop
+. AuthenticateRdsAccount -DeploymentUrl $RDBrokerURL -Credential $TenantAdminCredentials -ServicePrincipal:($isServicePrincipal -eq 'True') -TenantId $AadTenantId
+
+SetTenantGroupContextAndValidate -TenantGroupName $definedTenantGroupName -TenantName $TenantName
+
+# Getting fqdn of rdsh vm
+$SessionHostName = (Get-WmiObject win32_computersystem).DNSHostName + "." + (Get-WmiObject win32_computersystem).Domain

```

Comment:
This code is also in Script-RegisterSessionHost.ps1 so it would be nice to centralize this into a function, but is not a big deal.

Reply:
good idea